### ISSUE TITLE
Upgrade axios to latest version

### ISSUE THREAD

**Issue by AuthorProxy** [aws-amplify/amplify-js#11987]
### Is this related to a new or existing framework?
No response

### Is this related to a new or existing API?
No response

### Is this related to another service?
No response

### Describe the feature you'd like to request
Upgrade axios to latest version

### Describe the solution you'd like
Upgrade axios to latest version

### Describe alternatives you've considered
Upgrade axios to latest version

### Additional context
No response

### Is this something that you'd be interested in working on?

- [x] 👋 I may be able to implement this feature request

**Comment by nadetastic** [comment]
Hi @AuthorProxy thanks for opening this feature request. Updating the version of axios is something that we are discussing internally and will provide a follow up soon.

Also, I have re-added the additional fields from the issue template as well to match with other FR in our repo 😄

**Comment by milotoor** [comment]
This feature request just gained a lot of urgency with this [CSRF vulnerability report from Snyk](https://security.snyk.io/vuln/SNYK-JS-AXIOS-6032459). See the [issue in axios's repo here](https://github.com/axios/axios/issues/6022). It's not clear to me that any version of axios does not suffer from this, so upgrading to v1 may or may not be helpful in this regard. But assuming a patch is incoming soon, an upgrade of some kind will be necessary from `aws-amplify` since [version 0.26.0 is pinned](https://github.com/aws-amplify/amplify-js/blob/main/packages/api-rest/package.json#L50) in `api-rest`.

**Comment by milotoor** [comment]
Axios has now patched the issue in [v1.6.0](https://github.com/axios/axios/releases/tag/v1.6.0)

**Comment by nadetastic** [comment]
Hi @milotoor thank you for sharing this! - The team is currently working to address this and will provide some updates soon.

**Comment by nathfy** [comment]
As a temporary work around you can add:
```
"overrides": {
    "axios": "^1.6.0"
  },
  "resolutions": {
    "axios": "^1.6.0"
  },
```
To your `package.json`.

**Comment by elorzafe** [comment]
@nathfy @AuthorProxy @nadetastic @milotoor 
Latest version of `aws-amplify@5.3.12` is now using`axios@1.6.0`

Thanks!

**Comment by cwomack** [comment]
This feature request to upgrade axios to v1.6.0 has been addressed in v5 of Amplify with the release of v5.3.12.  We recommend that you upgrade to the latest version of `aws-amplify` to address this issue.

https://www.npmjs.com/package/aws-amplify/v/5.3.12

**Comment by cwomack** [comment]
Hello @AuthorProxy,

Thank you for reaching out regarding the Axios package flagged by your report. On November 6, 2023, Amplify has published a new version of the aws-amplify package, version 5.3.12, which upgrades our Axios dependency from 0.26.0 to 1.6.0. We recommend that you upgrade to the latest version of aws-amplify to address this issue.

https://www.npmjs.com/package/aws-amplify/v/5.3.12
